### PR TITLE
Descend into TermBinds in `beta`

### PR DESCRIPTION
**Pull request by zliu41**
Previously `beta` didn't recursively process the term after creating a `TermBind` (https://github.com/input-output-hk/plutus/pull/4592#discussion_r866666660).

### CODE REVIEW

**Comment by michaelpj** [comment]
So the thing I was wondering is: can we just write the whole thing using `transformOf`? That's what we do elsewhere for context-insensitive transformations, see e.g. `LetMerge`. 

I think we should be able to, and that should deal with this more transparently.

**Review by effectfully — COMMENTED** [review]
I don't know much about our approach here, hence no real review from me.

But we should have a test for `TyInst`-`TyAbs` too.

**Comment by effectfully** [comment]
@zliu41 ah, sorry, I saw changes in handling of `TyInst _ (TyAbs a n k body) tyArg`, so I thought that some test should change too, but I now realize those changes were merely formatting, so ignore my comment.

**Comment by zliu41** [comment]
> can we just write the whole thing using transformOf? 

@michaelpj I doubt we can, because we'd run into the problem described in `Note [Multi-beta]`:

> However, if we naively do a bottom-up pattern-matching transformation on the AST to look for immediately-applied lambda abstractions then we will get the following:
> ...

That's probably why `beta` uses a top-down approach.

**Comment by michaelpj** [comment]
Hmm yes you're right. A bit unsatisfying.

File: plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs [hunk 1]
```diff
@@ -124,9 +125,14 @@ beta
 beta = \case
     -- See Note [Multi-beta]
     -- This maybe isn't the best annotation for this term, but it will do.
-    (extractBindings -> Just (bs, t)) -> Let (termAnn t) NonRec bs (beta t)
+    (extractBindings -> Just (bs, t)) ->
```
Comment (michaelpj) [inline plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs:128]: Here's a possible alternative that might be nicer. It makes the transformation part simpler and I think makes the pattern of "do the local transform and then recurse" clearer.

```
localTransform = \case
  (extractBindings -> ...
  TyInst ...
  t -> t
beta t = (localTransform t) & termSubterms %~ beta
```

Comment (zliu41) [reply plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs]: Good suggestion. Updated.

**Review by michaelpj — APPROVED** [review]
lovely

**Pull request merged**